language/compiler/ir_to_bytecode/syntax/src/syntax.lalrpop

```diff
@@ -343,14 +345,61 @@ FunctionBlock: (Vec<(Var_, Type)>, Block) = {
     "{" <locals: Declarations> <stmts: Statements> "}" => (locals, Block::new(stmts))
 }
 
+KindConstraint: KindConstraint = {
+    <n: Name> => KindConstraint::new(n, Kind::All),
+    <n: Name> ":" "all" => KindConstraint::new(n, Kind::All),
+    <n: Name> ":" "resource" => KindConstraint::new(n, Kind::Resource),
+    <n: Name> ":" "unrestricted" => KindConstraint::new(n, Kind::Unrestricted),
+}
+
+KindConstraintsImpl: KindConstraints = {
+    "<" <constraints: Comma<KindConstraint>> ">" => KindConstraints::new(constraints)
```

Comment:
The lexer seems to be doing something weird here. "<T>", "<  T  >", "< >" are each considered a single token. @tnowacki could you please have a look? Thanks!

Some alternatives do work: "[T>", "<T]", "[T]", "[  T  ]". Not sure what this implies though.

Reply:
Resolved. We have a regex "<..*>" which can be removed with @sblackshear's recent removal of assume/verify statements.